10up/ElasticPress
# Display grouped error messages with solutions.

### Description of the Change
Updates the Errors tab of the Sync page to display error messages grouped by type, with a description that links to supporting documentation.

![elasticpress test_wp-admin_admin php_page=elasticpress-sync](https://github.com/10up/ElasticPress/assets/4100733/58cd98eb-ff16-4936-9cf2-1269ef3b94c6)


Closes #2847

### How to test the Change
Use this filter to cause errors to appear.

```
add_filter( 'ep_total_field_limit', function() { return 100; } );

add_filter(
	'ep_prepare_meta_data',
	function ( $post_meta, $post ) {
		if ( 0 === $post->ID % 2 ) {
			for ( $i = 0; $i < 100; $i++ ) {
				$post_meta[ "test_meta_{$i}_title_{$post->ID}" ] = 'Lorem';
				$post_meta[ "test_meta_{$i}_body_{$post->ID}" ]  = 'Ipsum';
			}
		}
		return $post_meta;
	},
	10,
	2
);
```

The exact number may need to be reduce to cause an error, depending on your data.

Run a sync, the errors tab should display a result similar to the screenshot.

Run another sync, the count for any repeated errors should increase.

Clear the log, the errors should be cleared as well.

### Changelog Entry
Added - A new Sync errors tab, with errors grouped by type and links to support documentation when available.

### Credits

Props @JakePT @apurvrdx1


### Checklist:
